File: UtilitiesVB/Base Functionality Extensions/ArrayIsAllocated.vb
Imports System.Runtime.CompilerServices

Public Module ArrayIsAllocated
    <Extension()>
    Public Function IsAllocated(ByVal inArray As System.Array) As Boolean
        Dim FlagEx As Boolean = True
        Try
            If inArray Is Nothing Then
                FlagEx = False
            ElseIf inArray.Length <= 0 Then
                FlagEx = False
            ElseIf inArray(0) Is Nothing Then
                FlagEx = False
            End If
        Catch ex As Exception
            FlagEx = False
        End Try
        Return FlagEx
    End Function
End Module

File: UtilitiesVB/Base Functionality Extensions/ArrayIsAllocated.cs
using System;

namespace UtilitiesVB
{

    public static class ArrayIsAllocated
    {
        public static bool IsAllocated(ref Array inArray)
        {
            bool FlagEx = true;
            try
            {
                if (inArray is null)
                {
                    FlagEx = false;
                }
                else if (inArray.Length <= 0)
                {
                    FlagEx = false;
                }
                else if (inArray.GetValue(0) == null)
                {
                    FlagEx = false;
                }
            }
            catch 
            {
                FlagEx = false;
            }
            return FlagEx;
        }

        public static bool IsAllocated(ref string[] inArray)
        {
            bool FlagEx = true;
            try
            {
                if (inArray is null)
                {
                    FlagEx = false;
                }
                else if (inArray.Length <= 0)
                {
                    FlagEx = false;
                }
                else if (inArray[0] is null)
                {
                    FlagEx = false;
                }
            }
            catch (Exception ex)
            {
                FlagEx = false;
            }
            return FlagEx;
        }
    }
}